### PR TITLE
1.19.3

### PR DESCRIPTION
**Pull request by basaigh**

### CODE REVIEW

File: src/main/java/com/github/steveice10/mc/protocol/codec/MinecraftCodecHelper.java [hunk 1]
```diff
@@ -696,6 +699,85 @@ public void writeChunkSection(ByteBuf buf, ChunkSection section) {
         this.writeDataPalette(buf, section.getBiomeData());
     }
 
+	public <E extends Enum<E>> EnumSet<E> readEnumSet(ByteBuf buf, Class<E> enumClass) {
+		E[] values = enumClass.getEnumConstants();
```
Comment (Camotoy) [inline src/main/java/com/github/steveice10/mc/protocol/codec/MinecraftCodecHelper.java]: Perhaps we can make this more performant if we instead reference `EnumClass.VALUES`?  `getEnumConstants` will clone the enum array. Ping me on Discord if this doesn't make sense.

File: src/main/java/com/github/steveice10/mc/protocol/data/MagicValues.java [hunk 2]
```diff
@@ -408,10 +408,11 @@ public class MagicValues {
         register(Advancement.DisplayData.FrameType.GOAL, 2);
 
         register(PlayerListEntryAction.ADD_PLAYER, 0);
-        register(PlayerListEntryAction.UPDATE_GAMEMODE, 1);
-        register(PlayerListEntryAction.UPDATE_LATENCY, 2);
-        register(PlayerListEntryAction.UPDATE_DISPLAY_NAME, 3);
-        register(PlayerListEntryAction.REMOVE_PLAYER, 4);
+        register(PlayerListEntryAction.INITIALIZE_CHAT, 1);
```
Comment (Camotoy) [inline src/main/java/com/github/steveice10/mc/protocol/data/MagicValues.java]: Let's take this opportunity to move this list away from MagicValues?

File: src/main/java/com/github/steveice10/mc/protocol/data/game/PlayerListEntryAction.java [hunk 3]
```diff
@@ -2,8 +2,9 @@
 
 public enum PlayerListEntryAction {
     ADD_PLAYER,
-    UPDATE_GAMEMODE,
-    UPDATE_LATENCY,
-    UPDATE_DISPLAY_NAME,
-    REMOVE_PLAYER;
+    INITIALIZE_CHAT,
+	UPDATE_GAME_MODE,
```
Comment (Camotoy) [inline src/main/java/com/github/steveice10/mc/protocol/data/game/PlayerListEntryAction.java]: Indent error?

File: src/main/java/com/github/steveice10/mc/protocol/packet/ingame/serverbound/ServerboundChatPacket.java [hunk 4]
```diff
@@ -21,52 +19,35 @@ public class ServerboundChatPacket implements MinecraftPacket {
     private final @NotNull String message;
     private final long timeStamp;
     private final long salt;
-    private final byte @NotNull[] signature;
-    private final boolean signedPreview;
-    private final List<LastSeenMessage> lastSeenMessages;
-    private final @Nullable LastSeenMessage lastReceivedMessage;
+    private final byte @Nullable[] signature;
+    private final int offset;
+	private final BitSet acknowledgedMessages;
```
Comment (Camotoy) [inline src/main/java/com/github/steveice10/mc/protocol/packet/ingame/serverbound/ServerboundChatPacket.java]: Indent error?

File: src/main/java/com/github/steveice10/mc/protocol/packet/ingame/serverbound/ServerboundChatPacket.java [hunk 5]
```diff
@@ -21,52 +19,35 @@ public class ServerboundChatPacket implements MinecraftPacket {
...
+			this.signature = null;
+		}
 
-        if (in.readBoolean()) {
-            this.lastReceivedMessage = new LastSeenMessage(helper.readUUID(in), helper.readByteArray(in));
-        } else {
-            this.lastReceivedMessage = null;
-        }
+        this.offset = helper.readVarInt(in);
+		this.acknowledgedMessages = helper.readFixedBitSet(in, 20);
```
Comment (Camotoy) [inline src/main/java/com/github/steveice10/mc/protocol/packet/ingame/serverbound/ServerboundChatPacket.java]: Indent error

**Review by Camotoy — COMMENTED** [review]
I think there's a lot of places where you're using a different indent than the rest of the project?

**Comment by basaigh** [comment]
> I think there's a lot of places where you're using a different indent than the rest of the project?

Ugh, sorry, thought I'd fixed that

**Comment by basaigh** [comment]
Will close, as has been manually merged.

**Pull request closed without merging**